tap_klaviyo_custom/streams.py

```diff
@@ -1,46 +1,214 @@
 """Stream type classes for tap-klaviyo-custom."""
 
+import base64
+import requests
 from pathlib import Path
-from typing import Any, Dict, Optional, Union, List, Iterable
+from typing import Any, Dict, Optional, Union, List, Iterable, cast
+import copy
+import urllib
+import json
+import time
 
 from singer_sdk import typing as th  # JSON Schema typing helpers
 
-from tap_klaviyo_custom.client import klaviyo-customStream
+from singer_sdk.helpers.jsonpath import extract_jsonpath
+from singer_sdk.streams import RESTStream
+import tap_klaviyo_custom.tap
+
+import singer
+
+LOGGER = singer.get_logger()
 
-# TODO: Delete this is if not using json files for schema definition
 SCHEMAS_DIR = Path(__file__).parent / Path("./schemas")
-# TODO: - Override `UsersStream` and `GroupsStream` with your own stream definition.
-#       - Copy-paste as many times as needed to create multiple stream types.
 
 
-class UsersStream(klaviyo-customStream):
+class ListsStream(RESTStream):
     """Define custom stream."""
-    name = "users"
-    path = "/users"
-    primary_keys = ["id"]
+    name = "lists"
+    path = "lists"
+    primary_keys = ["list_id"]
     replication_key = None
-    # Optionally, you may also use `schema_filepath` in place of `schema`:
-    # schema_filepath = SCHEMAS_DIR / "users.json"
-    schema = th.PropertiesList(
-        th.Property("name", th.StringType),
-        th.Property("id", th.StringType),
-        th.Property("age", th.IntegerType),
-        th.Property("email", th.StringType),
-        th.Property("street", th.StringType),
-        th.Property("city", th.StringType),
-        th.Property("state", th.StringType),
-        th.Property("zip", th.StringType),
-    ).to_dict()
-
-
-class GroupsStream(klaviyo-customStream):
+    schema_filepath = SCHEMAS_DIR / "lists.json"
+
+    url_base = "https://a.klaviyo.com/api/v2/"
+
+    records_jsonpath = "$[*]"  # Or override `parse_response`.
+
+    def get_url_params(self, partition: Optional[dict]) -> Dict[str, Any]:
+        """Return a dictionary of values to be used in URL parameterization."""
+        params = {}
+        params.update({"api_key": self.config['api_key']})
+        return params
+
+    def prepare_request(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> requests.PreparedRequest:
+        """Prepare a request object.
```

Comment:
Are these boiler point comments or are you adding them?

Reply:
Those are all boilerplate comments so it's easy to understand what the different Singer SDK methods are doing. If you think they're unnecessary, I'll remove them. I just thought having descriptors of what each method does might be helpful in the long run

Reply:
Since all taps use this sdk i feel it would unnecessary to have the same comments in every tap built based on the sdk.

Reply:
I don't know if I agree. The `"""..."""` is Python's way of doing method docs. Sure the behavior of these methods comes from the SDK but the SDK has some weak docs currently, and the comments are helping us remember how the SDK works. They're also generated boilerplate so it's not like we really needed to write or maintain them